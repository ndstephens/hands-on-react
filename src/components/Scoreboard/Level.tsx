import styled from '@emotion/styled';

interface Props {
  /**
   * Array of possible game levels
   */
  children: string[];
}

export const Level = ({ children }: Props) => {
  return (
    <Select>
      {children.map((item: string) => (
        <Option key={item} value={item}>
          {item}
        </Option>
      ))}
    </Select>
  );
};

const Select = styled.select`
  margin: 0;
  height: 2.5vw;
  border-radius: 0;
  border: 0.15vw solid;
  border-color: white #9e9e9e #9e9e9e white;
  background-color: #d1d1d1;
`;

const Option = styled.option`
  font-weight: normal;
  display: block;
  white-space: nowrap;
  min-height: 1.2em;
  padding: 0 0.2vw 0.2vw;
`;
