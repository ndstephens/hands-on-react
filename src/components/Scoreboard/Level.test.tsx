import { render } from '@testing-library/react';

// import { render, screen } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import { Level } from './Level';

describe('Level test cases', () => {
  // Default props
  const children = ['beginner', 'intermediate', 'expert'];

  it('Level renders correctly', () => {
    // const onChange = jest.fn();

    const { asFragment } = render(<Level>{children}</Level>);

    expect(asFragment()).toMatchSnapshot();
  });

  // it('Select level behaviour', () => {
  //   const onChange = jest.fn();

  //   render(
  //     <Level onChange={onChange}>
  //       {['beginner', 'intermediate', 'expert']}
  //     </Level>
  //   );

  //   userEvent.selectOptions(screen.getByRole('combobox'), 'intermediate');

  //   expect(screen.getByRole('option', { name: 'intermediate' })).toBeEnabled();

  //   expect(onChange).toHaveBeenCalled();
  // });
});
