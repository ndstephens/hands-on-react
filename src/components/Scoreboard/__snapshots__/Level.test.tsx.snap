// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level test cases Level renders correctly 1`] = `
<DocumentFragment>
  <select
    class="css-55qvia"
  >
    <option
      class="css-v3kp7e"
      value="beginner"
    >
      beginner
    </option>
    <option
      class="css-v3kp7e"
      value="intermediate"
    >
      intermediate
    </option>
    <option
      class="css-v3kp7e"
      value="expert"
    >
      expert
    </option>
  </select>
</DocumentFragment>
`;
